JMESPath docs, unit test for jmes functions and merge with master (#1)
EchoShoot/parsel: README.rst

> Parsel is a BSD-licensed Python_ library to extract and remove data from HTML_
> and XML_ using XPath_ and CSS_ selectors, as well as JSON_ using JMESPath_, optionally combined with
> `regular expressions`_.

Reviewer: What about:

```suggestion
Parsel is a BSD-licensed Python_ library to extract and remove data from HTML_, 
JSON, and XML_ documents using expressions written in CSS_, JMESPath_, or 
XPath_, optionally combined with `regular expressions`_. 
```

or, to make things more clear while keeping the first paragraph short and to the point:

```suggestion
Parsel is a BSD-licensed Python_ library to extract data from HTML_, JSON, and 
XML_ documents.

It supports:

-   CSS_ and XPath_ expressions for HTML and XML documents

-   JMESPath_ expressions for JSON documents

-   `Regular expressions`_

-   Removing data
```

Author: Later I will accept the second approach, but before I wanted to mention that removing data does not seem like a major feature compared to the other 3 mentioned. I would remove it from the first contact the user has with the tool and put it further down.

Reviewer: I am fine with that.

    'Hello, Parsel!'
    >>> html_selector.xpath('//h1/text()').re(r'\w+')
    ['Hello', 'Parsel']
    >>> for li in selector.css('ul > li'):
    ...     print(li.xpath('.//@href').get())
    http://example.com
    http://scrapy.org

    >>> json_text = """
            {
                "page_title": "Hello, Parsel!",
                "links": [
                    {"url": "http://example.com", "title": "Link 1"},
                    {"url": "http://scrapy.org", "title": "Link 2"}
                ]
            }"""
    >>> json_selector = Selector(text=json_text)
    >>> json_selector.jmespath("page_title").get()
    'Hello, Parsel!'
    >>> json_selector.jmespath("links[*].title").getall()
    ['Link 1', 'Link 2']

.. _CSS: https://en.wikipedia.org/wiki/Cascading_Style_Sheets
.. _HTML: https://en.wikipedia.org/wiki/HTML
.. _JSON: https://en.wikipedia.org/wiki/JSON
.. _open online demo: https://colab.research.google.com/drive/149VFa6Px3wg7S3SEnUqk--TyBrKplxCN#forceEdit=true&sandboxMode=true
.. _Python: https://www.python.org/
.. _regular expressions: https://docs.python.org/library/re.html
.. _XML: https://en.wikipedia.org/wiki/XML
.. _XPath: https://en.wikipedia.org/wiki/XPath

> .. _JMESPath: https://jmespath.org/

Reviewer: Maybe keep the alphabetical sorting?

Author: Oh I was thinking the order was about scopes.
Things that will be extracted.
Others.
Selectors.

I will adjust this

Author: Also in the revision above you don't mention _JSON, should we delete it then? the ref

Reviewer: I just forgot to put it in the JSON reference, but I think it would be would to keep the link.
